examples/customColorLayerEffect.html

```diff
@@ -0,0 +1,140 @@
+                        for (int j = 0; j < 3; j++) {
+                            result += kernel[i][j]*image[i][j];
+                        }
+                    }
+                    return result;
+                }
+
+                float edge(sampler2D textu, float stepx, float stepy, vec2 center){
+                    // get samples around pixel
+                    mat3 image = mat3(length(texture2D(textu,center + vec2(-stepx,stepy)).rgb),
+                                      length(texture2D(textu,center + vec2(0,stepy)).rgb),
+                                      length(texture2D(textu,center + vec2(stepx,stepy)).rgb),
+                                      length(texture2D(textu,center + vec2(-stepx,0)).rgb),
+                                      length(texture2D(textu,center).rgb),
+                                      length(texture2D(textu,center + vec2(stepx,0)).rgb),
+                                      length(texture2D(textu,center + vec2(-stepx,-stepy)).rgb),
+                                      length(texture2D(textu,center + vec2(0,-stepy)).rgb),
+                                      length(texture2D(textu,center + vec2(stepx,-stepy)).rgb));
+                    vec2 result;
+                    result.x = convolve(sobelKernelX, image);
+                    result.y = convolve(sobelKernelY, image);
+
+                    float color = clamp(length(result), 0.0, 255.0);
+                    return color;
+                }
+            `);
+
+            itowns.ShaderChunk.customBodyColorLayer(`
+                ivec2 textureSize2d = textureSize(tex,0);
+                vec2 resolution = vec2(float(textureSize2d.x), float(textureSize2d.y));
+
+                float step = layer.amount_effect;
+
+                vec2 cuv = pitUV(uv.xy, offsetScale);
+
+                float value = edge(tex, step/resolution.x, step/resolution.y, cuv);
+
+                color = vec4(vec3(0.27, 0.4, 0.42) *  value * 0.8, value);
+            `);
+
+            // Add a TMS imagery source
+            var opensmSource = new itowns.TMSSource({
+                isInverted: true,
+                // eslint-disable-next-line no-template-curly-in-string
+                url: 'http://c.tile.stamen.com/watercolor/${z}/${x}/${y}.jpg',
+                networkOptions: { crossOrigin: 'anonymous' },
+                extent,
+                crs: 'EPSG:3857',
+                attribution: {
+                    name: 'OpenStreetMap',
+                    url: 'http://www.openstreetmap.org/',
+                },
+            });
+
+            // Add a TMS imagery layer
+            var opensmLayer = new itowns.ColorLayer('OPENSM', {
+                updateStrategy: {
+                    type: itowns.STRATEGY_DICHOTOMY,
+                },
+                source: opensmSource,
+                type_effect: itowns.colorLayerEffects.customEffect,
+                amount_effect: 1.0,
+            });
+
+            view.addLayer(opensmLayer);
```

Comment:
it could be a good idea to add two layers in the view with the same source: one normal and one with the special effect.